src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/LicenceService.kt

```diff
@@ -49,9 +42,9 @@ class LicenceService(
     val isCas2Accepted = cas2Referral?.bassOffer?.bassAccepted == OfferAccepted.YES
 
     val address = when {
-      isCurfewApprovedPremisesRequired && !isCas2ApprovedPremisesRequired -> curfew.approvedPremisesAddress
-      isCas2ApprovedPremisesRequired -> cas2Referral.approvedPremisesAddress
-      isCas2Requested && isCas2Accepted -> cas2Referral.bassOffer
+      isCurfewApprovedPremisesRequired && !isCas2ApprovedPremisesRequired -> curfew?.approvedPremisesAddress
+      isCas2ApprovedPremisesRequired -> cas2Referral?.approvedPremisesAddress
+      isCas2Requested && isCas2Accepted -> cas2Referral?.bassOffer
       else -> proposedAddress?.curfewAddress
     }
 
```

Comment:
better to transform the object `transformToModelCurfewAddress` on the line below and then you don't need to handle the null above

`address?.let { formatAddress(it) }`

maybe you don't event need to `formatAddress` first?